controllers/tokenupload_controller.go

```diff
@@ -0,0 +1,229 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"time"
+
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	"github.com/go-logr/logr"
+
+	"github.com/redhat-appstudio/service-provider-integration-operator/pkg/spi-shared/tokenstorage"
+	"k8s.io/apimachinery/pkg/types"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	spi "github.com/redhat-appstudio/service-provider-integration-operator/api/v1beta1"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const tokenSeretLabel = "spi.appstudio.redhat.com/token"
+
+// TokenUploadReconciler reconciles a Secret object
+type TokenUploadReconciler struct {
+	client.Client
+	Scheme       *runtime.Scheme
+	TokenStorage tokenstorage.NotifyingTokenStorage
+}
+
+func (r *TokenUploadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	//_ = log.FromContext(ctx)
+
+	lg := log.FromContext(ctx)
+
+	list := &corev1.SecretList{}
+
+	err := r.List(context.TODO(), list, client.HasLabels{tokenSeretLabel})
+	if err != nil {
+		lg.Error(err, "Can not get list of secrets ")
+		return ctrl.Result{}, err
+	}
+
+	for _, s := range list.Items {
+
+		var accessToken spi.SPIAccessToken
+
+		// we immediatelly delete the Secret
+		err := r.Delete(context.TODO(), &s)
+		if err != nil {
+			logError(s, err, r, "can not delete the Secret ", lg)
+			return ctrl.Result{}, err
+		}
+
+		// if spiTokenName field is not empty - try to find SPIAccessToken by it
+		if len(s.Data["spiTokenName"]) > 0 {
+			accessToken = spi.SPIAccessToken{}
+			err = r.Get(context.TODO(), types.NamespacedName{Name: string(s.Data["spiTokenName"]), Namespace: s.Namespace}, &accessToken)
+
+			if err != nil {
+				logError(s, err, r, "can not find SPI access token "+string(s.Data["spiTokenName"]), lg)
+				return ctrl.Result{}, err
```

Comment:
Can we consider `continue` operator here? Otherwise we can be completely blocked by one "bad" secret.

Reply:
Well, it will not be blocked because the Secret is deleted so will not appear in the next reconcile cycle.
But, I agree, interrupting the loop this way is not nice and not optimal, thanks.